app/services/ranking_calculator_service.rb

```diff
@@ -0,0 +1,63 @@
+class RankingCalculatorService
+  class << self
+    def rank(game)
+      return unless game.finished?
+
+      winners = game.winning_team
+      losers = game.losing_team
+
+      update_player(winners[0], winners[1], losers[0], losers[1], 1)
+      update_player(winners[1], winners[0], losers[0], losers[1], 1)
+
+      update_player(losers[0], losers[1], winners[0], winners[1], 0)
+      update_player(losers[1], losers[0], winners[0], winners[1], 0)
+    end
+
+    private
+
+    def update_player(player, teammate, opponent_1, opponent_2, score)
+      # To account for the rank of the teammate, we sum the opponents then
+      #   subtract the rank of the teammate. This way, playing with a lower
+      #   ranked teammate counts as a more challenging game and vice versa.
+      o_rank = opponent_1.rank + opponent_2.rank - teammate.rank
+
+      player.update_attributes(rank: new_elo(player.rank, o_rank, score))
+    end
+
+    def new_elo(player, opponent, score)
```

Comment:
Would appending `_rank` make the variable names make more sense as they are not Player objects.